#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/lights/protos/CeilingLight.proto"

WorldInfo {
  info [
    "iRobot Create simulation"
    "The vacuum cleaner robot goes randomly into the apartment while a supervisor tracks its position and alterates the ground texture."
    "The robot can be stopped by the virtual walls emitting IR light."
  ]
  title "iRobot Create"
}
Viewpoint {
  orientation -0.5218586376281682 0.5506271962059177 0.651516118857609 2.0400139837873748
  position 0.7204735378443188 -3.3071967183524182 14.450025353335036
  near 0.1
}
TexturedBackground {
}
CeilingLight {
  translation 1.82639 -1.29685 2.4
  pointLightColor 1 1 0.9372549019607843
  pointLightIntensity 7
}
CeilingLight {
  translation -3.25511 1.9443400000000004 2.4
  name "ceiling light(2)"
  pointLightIntensity 6
}
DEF diff_robot Robot {
  translation -4.972500780240526 -0.09813434682481352 0.039720981511511884
  rotation -0.6277219203741203 0.7784376601127584 -2.226230178976761e-08 3.1415926169366926
  children [
    GPS {
    }
    InertialUnit {
      name "imu"
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 5752.613440718438
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.05 0.06 0
        rotation 0.12665498351696916 -0.7014110424241504 0.7014136188553677 2.889625202382944
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 6161.739816563218
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.05 -0.06 0
        rotation -0.3889860546326174 0.6514164305149865 -0.6514188233053018 3.8835415462823986
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 5752.835684820817
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.05 0.06 0
        rotation 0.20629640415541015 -0.6918953232797053 0.6918978647576676 2.734709309203245
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position 6161.933027773686
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.05 -0.06 0
        rotation -0.4619860466534465 0.6271227963762102 -0.62712509993047 4.0071432516784515
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation 0.1 -0.03 0
      rotation 0 0 1 -0.3
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.1 0.03 0
      rotation 0 0 1 0.3
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
  ]
  name "robot(1)"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "point_follower"
}
Robot {
  children [
    Solid {
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "textures/logo.png"
              ]
            }
            metalness 0
          }
          geometry Plane {
            size 10 10
          }
        }
      ]
    }
    Display {
      translation 0 0 0.01
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0.6 1
            baseColorMap ImageTexture {
              url [
                "textures/sin.png"
              ]
            }
            roughness 0.5
            metalness 0
          }
          geometry Plane {
            size 9.9 9.9
          }
        }
      ]
      name "ground_display"
      width 500
      height 500
    }
  ]
  boundingObject Plane {
    size 9.9 9.9
  }
  locked TRUE
  controller "ground"
  supervisor TRUE
}
